pkg/api/v1/resource/helpers.go

```diff
@@ -56,6 +56,21 @@ func podRequestsAndLimitsWithoutOverhead(pod *v1.Pod, reqs, limits v1.ResourceLi
 		maxResourceList(reqs, container.Resources.Requests)
 		maxResourceList(limits, container.Resources.Limits)
 	}
+
+	// If any container does not specify a limit that means the resource is not limited for the pod as a whole.  The
+	// previously calculated max limit would only be the max of the explicitly set limits.  However, an unspecified
+	// limit should be considered greater than any explicitly set limit, so delete the calculated limit if there is one.
```

Comment:
@alculquicondor Is this how you expect limits to accumulate?

Reply:
The scheduler doesn't really pay attention to the limits (other than for the pod resource metric), so it is more of a question to the node team on how they set cgroups when it comes to limits, which I think is unlimited when not set.

Reply:
cc @smarterclayton who wrote the pod resource metric

Reply:
Hmm. Interesting. Chiming in as a sig-node reviewer here. Shouldn't this logic behave in the same way as how we find the pod limit/requests for the pod cgroup? That logic lives here; https://github.com/kubernetes/kubernetes/blob/3ffdfbe286ebcea5d75617da6accaf67f815e0cf/pkg/kubelet/cm/helpers_linux.go#L159-L172.

As I read it, this function (in this PR) is used for the pod sandbox `resources` defined here; https://github.com/kubernetes/cri-api/blob/master/pkg/apis/runtime/v1/api.proto#L368. And to me, the current behavior looks wrong. I need to look at this PR a bit deeper to understand if this is the correct solution or not, but it looks more correct than the current behavior in my opinion.